utils/CMakeLists.txt

```diff
@@ -48,7 +48,11 @@ target_sources(utils
     aws_sigv4.cc
     s3/aws_error.cc
     s3/client.cc
-    s3/retry_strategy.cc)
+    s3/retry_strategy.cc
+    s3/config_file_aws_credentials_provider.cc
```

Comment:
either drop them all into `credential_provider` directory and remove this suffix from all files, or make it a prefix instead of a suffix:
```bash
credentials_provider_aws_config_file.cc
credentials_provider_aws_environment.cc
credentials_provider_instance_profile.cc
credentials_provider_sts_assume_role.cc
```